tests/run-make/pgo-branch-weights/rmake.rs

```diff
@@ -0,0 +1,75 @@
+// This test generates an instrumented binary - a program which
+// will keep track of how many times it calls each function, a useful
+// feature for optimization. Then, an argument (aaaaaaaaaaaa2bbbbbbbbbbbb2bbbbbbbbbbbbbbbbcc)
+// is passed into the instrumented binary, which should react with a number of function calls
+// fully known in advance. (For example, the letter 'a' results in calling f1())
+
+// If the test passes, the expected function call count was added to the use-phase LLVM-IR.
+// See https://github.com/rust-lang/rust/pull/66631
+
+// FIXME(mati865): MinGW GCC miscompiles compiler-rt profiling library but with Clang it works
+// properly. Since we only have GCC on the CI ignore the test for now.
+
+//@ needs-profiler-support
+//@ ignore-windows-gnu
+//@ ignore-cross-compile
+
+use std::io::{self, Write};
+use std::process::Command;
+use run_make_support::{llvm_profdata, rustc, rustdoc, tmp_dir};
+
+fn main() {
+    // For some very small programs GNU ld seems to not properly handle
+    // instrumentation sections correctly. Neither Gold nor LLD have that
+    // problem.use run_make_support::{run, rustc, tmp_dir};
+    let common_flags = if env::consts::OS == "linux"
+        && env::var("TARGET").unwrap_or_default().contains("x86") {
+        "-Clink-args=-fuse-ld=gold" // FIXME, Why was there a comma at the start?
```

Comment:
Do you mean in `ifneq (,$(findstring x86,$(TARGET)))`? `findstring` returns `$(TARGET)` (non-empty) if it contains the substring `x86` or an empty string otherwise, so checking that `findstring` does not return an empty string checks that target contains the substring...

Reply:
Wew, had to read that one 5 times before I got it. But I get it, now!